packages/hooks/src/useFileUploader.ts

```diff
@@ -0,0 +1,98 @@
+import DocumentUploader from '@binary-com/binary-document-uploader';
+import { useMutation } from '@deriv/api';
+import { WS, compressImageFiles, readFiles } from '@deriv/shared';
+import { useCallback, useMemo, useState } from 'react';
+
+type TSettingsPayload = NonNullable<
+    NonNullable<NonNullable<Parameters<ReturnType<typeof useMutation<'document_upload'>>['mutate']>>[0]>['payload']
+>;
+
+type TFile =
+    | Partial<
+          TSettingsPayload & {
+              filename: File['name'];
+              buffer: FileReader['result'];
+              documentFormat: string;
+              file_size: File['size'];
+          }
+      >
+    | {
+          message: string;
+      };
+
+const fileReadErrorMessage = (filename: string) => {
+    return `Unable to read file ${filename}`;
+};
+
+/**
+ * Custom hook to handle file uploading with the binary-document-uploader package
+ */
+const useFileUploader = () => {
+    const [error, setError] = useState<unknown>(null);
+    const uploader_instance = useMemo(() => new DocumentUploader({ connection: WS.getSocket() }), []);
+
+    const upload = useCallback(
+        async (files: File[], settings?: TSettingsPayload, onError?: () => void) => {
+            if (!files?.length) return Promise.reject(new Error('No files selected'));
+
+            return new Promise<{
+                message?: string;
+                warning?: string;
+                [key: string]: any;
+            }>((resolve, reject) => {
+                let is_any_file_error = false;
+                let file_error: string | null = null;
+
+                compressImageFiles()
+                    .then((files_to_process: Blob[]) => {
```

Comment:
just a suggestion:
would it be more clear to refactor it with  try {} catch (e) {} and async await?

Reply:
tried to refactor using async/await and then we got `Promise executor functions should not be async.eslint[no-async-promise-executor](https://eslint.org/docs/rules/no-async-promise-executor)`

Reply:
Still I'll look into it, if we can move this to async/await